brave/src/main/java/com/linecorp/armeria/client/brave/ArmeriaHttpClientAdapter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.brave;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.internal.brave.SpanTags;
+
+import brave.http.HttpClientAdapter;
+
+public class ArmeriaHttpClientAdapter extends HttpClientAdapter<RequestLog, RequestLog> {
+    @Override
+    public String path(RequestLog requestLog) {
+        return requestLog.path();
+    }
+
+    @Override
+    public String method(RequestLog requestLog) {
+        return requestLog.method().name();
+    }
+
+    @Override
+    public String url(RequestLog requestLog) {
+        return SpanTags.generateUrl(requestLog);
+    }
+
+    @Override
+    @Nullable
+    public String requestHeader(RequestLog requestLog, String name) {
+        return requestLog.requestHeaders().get(name);
+    }
+
+    @Override
+    public Integer statusCode(RequestLog requestLog) {
+        return requestLog.status().code();
```

Comment:
do this instead, assuming code is integer

```java
  @Override @Nullable public Integer statusCode(RequestLog requestLog) {
    int result = statusCodeAsInt(requestLog);
    return result != 0 ? result : null;
  }

  @Override public int statusCodeAsInt(RequestLog requestLog) {
    Status status = requestLog.status(); // adrian doesn't know the actual type here and neither if it can be null
    return status != null ? status.code() : 0;
  }
```

Reply:
Due to armeria has unknown state,  I check unknown one.